src/extensions/default/Brackets-InteractiveLinter/main.js

```diff
@@ -145,13 +160,15 @@ define(function (require, exports, module) {
             addGutter(currentEditor);
             handleDocumentChange();
         }
+
     }
 
 
     /**
      * Function that gets called when Brackets is loaded and ready
      */
     function appReady() {
+        loadPreferences(); 
         // Removes the default Brackets JSLint linter
```

Comment:
Because we are going to disable everything by default, and because it will add a lot of network size when we first load the app, I think we should do lazy-loading of all the linting plugins. To do this will requires a few things:

* Move all of the code from 172 to 205 into a function named `init`.  This `init` function will take care of loading all the plugins and registering them, as well as calling `setDocument` to enable any in the current editor.
* After you call `loadPreferences()` on 171, do an `if` check to see if the `enabled` pref is true, and if it is, call `init()` immediately (i.e., if the user has the pref `enabled=true`, then we'll load all the plugins right now on startup).  If it's `false`, add a one time event listener (e.g., the `one()` method) for the `enabled` pref changing to true, and call `init()` when that happens.  This way we can wait until they opt-in to having these linters turned on before we do the step of loading everything over the network.

Does this make sense?  When you're done doing all this, you should be able to test both ways of loading things by setting your default value for `enabled` above to `true` or `false` and then watching your dev tools Network tab to see what does and doesn't load in each case.